### ISSUE TITLE
DB should abort enqueue query for non-existent queue rather than die

### ISSUE THREAD

**Issue by plajjan** [actonlang/acton#483]
This is a followup from #472, where a bug in the RTS meant we tried to enqueue on a queue that did not exist. As a result, the DB died with an assert. It should rather handle that gracefully by aborting the transaction and continuing to live and serve other requests. It should also return a proper error message to the client.

The DB code is sprinkled with asserts, so there are probably many cases like this but it feels like we can start to pick them off as we encounter them.

**Comment by aagapi** [comment]
I agree the assert should be removed, and an error returned to the caller. It is then that caller's responsibility to either abort the transaction, create that queue or do something else in the same transaction and choose to commit.
This is in line with the current API design, whereby the DB itself only aborts transactions due to concurrency errors (of which DB clients later learn upon a commit attempt), whereas logical errors (such as non existing queues, tables etc) are merely statused back to DB clients, that then choose what to do with this information (including aborting explicitly if they so choose).
Good find, I will implement the above behavior as well as handle other asserts similarly if I find any. Assigning to myself.
